Assets/Scripts/Characters/EntityController.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EntityController : MonoBehaviour {
+
+    public float health { get; set; }
+    public float totalHealth { get; set; }
+
+    public bool immobile { get; set; }
+    public bool fixedFacing { get; set; }
+    public bool silent { get; set; }
+    public bool cantAttack { get; set; }
+
+    public float endStun;
+    public float endFreeze;
+
+    public IDictionary<Stat, float> baseStats;
+    public List<Effect> currentEffects;
+    public IDictionary<Stat, List<Modifier>> activeModifiers;
+    public IDictionary<Stat, float> effectiveStats;
+
+    protected Rigidbody charRigidbody;
+    protected Vector3 movement;
+
+    // Use this for initialization
+    protected virtual void Start () {
+        this.charRigidbody = GetComponent<Rigidbody> ();
+        this.immobile = false;
+        this.fixedFacing = false;
+        this.silent = false;
+        this.cantAttack = false;
+        this.currentEffects = new List<Effect> ();
+        this.baseStats = new Dictionary<Stat, float> ();
+        this.effectiveStats = new Dictionary<Stat, float> ();
+        this.activeModifiers = new Dictionary<Stat, List<Modifier>> ();
+        foreach (Stat s in System.Enum.GetValues(typeof(Stat))) {
+            baseStats [s] = 0f;
+            effectiveStats [s] = 0f;
+            activeModifiers [s] = new List<Modifier> ();
+        }
+    }
+
+    // Update is called once per frame
+    protected virtual void Update () {
```

Comment:
Pull the removal of expired effects and the calculation of stats into two private methods?

```
void Update() {
  removeExpiredEffects();
  calculateEffectiveStats();
  if (isDead()) {
    ...
  }
  ...
}
```